src/backwardproblem.cpp

```diff
@@ -42,51 +45,63 @@ void BackwardProblem::workBackwardProblem() {
         model->nplist() == 0) {
         return;
     }
-    
+
     int it = rdata->nt - 1;
     model->initializeB(xB, dxB, xQB);
-    handleDataPointB(it);
-    solver->setupB(&which, rdata->ts[it], model, xB, dxB, xQB);
-    
-    --it;
-    --iroot;
-
-    while (it >= 0 || iroot >= 0) {
-
-        /* check if next timepoint is a discontinuity or a data-point */
-        double tnext = getTnext(discs, iroot, it);
 
-        if (tnext < t) {
-            solver->runB(tnext);
-            solver->writeSolutionB(&t, xB, dxB, xQB, this->which);
-            solver->getDiagnosisB(it, rdata, this->which);
-        }
-
-        /* handle discontinuity */
-        if (model->ne > 0 && rdata->nmaxevent > 0 && iroot >= 0) {
-            if (tnext == discs.at(iroot)) {
-                handleEventB(iroot);
-                --iroot;
-            }
-        }
-
-        /* handle data-point */
-        if (tnext == rdata->ts[it]) {
-            handleDataPointB(it);
-            it--;
-        }
-
-        /* reinit states */
-        solver->reInitB(which, t, xB, dxB);
-        solver->quadReInitB(which, xQB);
+    if (std::isinf(model->getTimepoint(it)) && solver->getNewtonSolverBackward())
+    {
+        SteadystateProblem sstate(solver, AmiVector(slice(rdata->x, it, rdata->nx)));
+        sstate.workSteadyStateBackwardProblem(rdata, solver, edata, model);
+        sstate.writeSolutionBackward(xQB);
+        --it;
     }
 
-    /* we still need to integrate from first datapoint to tstart */
-    if (t > model->t0()) {
-        /* solve for backward problems */
-        solver->runB(model->t0());
-        solver->writeSolutionB(&t, xB, dxB, xQB, this->which);
-        solver->getDiagnosisB(0, rdata, this->which);
+    if (it>=0 && model->getTimepoint(it) > model->t0())
```

Comment:
What does `model->getTimepoint(it)` do if it == -1? Does it return false?
It boils down to some `amivector.at(-1)`, which should be failsafe... 
You have tested this, @paszkow ? If it works, I'm fine.

Reply:
The timepoints are stored in std::vector which at method accepts values of size_t type. I am not a fan of implicit conversions because you never know what may happen. Safer to keep it this way.